nxc/protocols/smb.py

```diff
@@ -978,6 +983,98 @@ def enumerate_sessions_info(self, sessions):
             except SessionError:
                 self.logger.fail("RDP is probably not enabled, cannot list remote IPv4 addresses.")
 
+    @requires_admin
+    def lsass(self):
+        def print_credentials(domain, username, password, lmhash, nthash):
+            if password is None:
+                password = ":".join(h for h in [lmhash, nthash] if h is not None)
+            self.logger.highlight(f"{domain}\\{username} {password}")
+
+        def save_credentials(domain, username, password, lmhash, nthash):
+            host_id = self.db.get_hosts(self.host)[0][0]
+            if password is not None:
+                credential_type = "plaintext"
+            else:
+                credential_type = "hash"
+                password = ":".join(h for h in [lmhash, nthash] if h is not None)
+            self.db.add_credential(credential_type, domain, username, password, pillaged_from=host_id)
+
+        def process_credentials(credentials):
+            if len(credentials) == 0:
+                self.logger.display("No credentials found")
+            credz_bh = []
+            domain = None
+            for cred in credentials:
+                if cred["domain"] is None:
+                    cred["domain"] = ""
+                domain = cred["domain"]
+                if "." not in cred["domain"] and cred["domain"].upper() in self.domain.upper():
+                    domain = self.domain
+                save_credentials(cred["domain"], cred["username"], cred["password"], cred["lmhash"], cred["nthash"])
+                print_credentials(cred["domain"], cred["username"], cred["password"], cred["lmhash"], cred["nthash"])
+                credz_bh.append({"username": cred["username"].upper(), "domain": domain.upper()})
+                add_user_bh(credz_bh, domain, self.logger, self.config)
+
+        session = Session()
```

Comment:
why not use `self.conn` and create another session when we already have one from nxc ?

Reply:
My bad ;)